framework/db/conditions/InConditionBuilder.php

```
@@ -153,79 +140,102 @@ protected function buildValues(ConditionInterface $condition, $values, &$params)
+                if (
+                    (is_array($value) || $value instanceof ArrayAccess)
+                    && $columnKey !== null
+                    && isset($value[$columnKey])
+                ) {
+                    $columnValue = $value[$columnKey];
+
+                    if ($columnValue instanceof ExpressionInterface) {
+                        $placeholder = $this->queryBuilder->buildExpression($columnValue, $params);
+                    } else {
+                        $placeholder = $this->queryBuilder->bindParam($columnValue, $params);
+                    }
```

PHPMD (2.15.0)

[error] The method buildCompositeInCondition uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)